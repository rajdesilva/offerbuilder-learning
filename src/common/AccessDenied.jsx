import { Button } from "antd";
import React from "react";
import { FormattedMessage } from "react-intl";
import { useLocation } from "react-router-dom";
import { baseURL } from "../helpers";
import styles from './css/AccessDenied.module.less';

export function AccessDenied() {
  const query = new URLSearchParams(useLocation().search);
  const emailId = query.get("email");
  return (
    <div className={styles.accessDenied}>
      <div className={styles.screenTitle}>
        <FormattedMessage id="nemo.accessDenied" />
      </div>
      <div className={styles.deniedMsg}>
        <FormattedMessage
          id="nemo.notHavAccessToApplication"
          values={{
            email: emailId,
          }}
        />
        <br />
        <br />
        <FormattedMessage id="nemo.contactSystmeAdmin" />
      </div>
      <Button
        type="primary"
        data-testid="denied-logout-btn"
        className={styles.deniedLogoutBtn}
        onClick={() => (window.location.href = `${baseURL.API_URL_BASE}logout`)}
      >
        <FormattedMessage id="nemo.logout" />
      </Button>
    </div>
  );
}
