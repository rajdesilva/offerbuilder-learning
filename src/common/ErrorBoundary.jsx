import React from "react";
import { FormattedMessage } from "react-intl";
import styles from './css/ErrorBoundary.module.less';
class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true });
    window.apm.captureError(error);
    window.apm.captureError(info);
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className={styles.errorBoundary}>
          <h2>
            <FormattedMessage id="nemo.somethingWentWrongPleaseTryAgain" />
          </h2>
        </div>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
