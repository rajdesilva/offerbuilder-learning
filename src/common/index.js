export { default as Loader } from "./Loader";
export { appConstants } from "./Constants";
export { appConstants_results } from "./Results_Constants";
export { AccessDenied } from "./AccessDenied";
export { UserConfirmation } from "./UserConfirmation";
