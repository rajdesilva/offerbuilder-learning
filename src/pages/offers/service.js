import { baseURL } from "../../helpers";
import {
  getParsedOffers,
  getSearchAndFilterImagesQueryUrl,
  getSearchAndFilterQueryURL,
  updateReduxWithOfferData,
} from "../../helpers/utility";
import { store } from "../../redux/store";
import {
  marketingActions,
  newOfferActions,
  offerListSearchAndFilterActions,
} from "./actions";
import { offerImageSearchAndFilterActions } from "./actions/offerImageSearchAndFilterActions";

export const validateOfferId = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/exist`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCloudinarySignature = () => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}cloudinary/signature`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCityFromLatLong = (latLong) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latLong.lat},${latLong.lng}&sensor=true&key=AIzaSyDydZJ7RZF1SAXwg4pZqnFFskYL8CW5RqA`;
  return fetch(url)
    .then((response) => response.json())
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error;
    });
};

export const getLanguages = () => {
  store.dispatch({
    type: marketingActions.FETCH_LANGUAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}languages`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_SUCCESS,
          payload: window.getValue(response, "data.languages"),
        });
      } else {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: marketingActions.FETCH_LANGUAGE_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const createNewOffer = (newOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers`;
  store.dispatch({
    type: newOfferActions.CREATE_NEW_OFFER_LOADING,
    payload: true,
  });
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const saveEditedOffer = (updatedOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers/${updatedOfferInfo.id}`;
  return window
    .fetchWrapper(url, {
      method: "PATCH",
      body: JSON.stringify(updatedOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const getOfferDetails = (offerId) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_DETAILS_LOADING,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`)
    .then((response) => {
      if (response.success) {
        let offerData = window.getValue(response, "data.offer");
        offerData = getParsedOffers(offerData);
        updateReduxWithOfferData(offerData);
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_SUCCESS,
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
        });
      }
    })
    .catch((error) => {
      console.log(error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
      });
    });
};

export const deleteOffer = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`, {
      method: "DELETE",
    })
    .then((response) => response);
};

export const updateOffer = (offerId, status) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/${status}`, {
      method: "PUT",
    })
    .then((response) => response);
};

export const searchAndFilterOffer = (offerType) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}offers/search/` +
        getSearchAndFilterQueryURL(offerType)
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
        payload: [],
      });
    });
};

export const getImagesFromProperties = (propertyCode) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}search?propertyCode=${propertyCode}`)
    .then((response) => response);
};

export const searchAndFilterImages = () => {
  store.dispatch({
    type: offerImageSearchAndFilterActions.OFFER_IMAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}images/search` +
        getSearchAndFilterImagesQueryUrl()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
        payload: [],
      });
    });
};

export const uploadImage = (file) => {
  let myHeaders = new Headers();

  myHeaders.append("Accept", "application/json");
  let formdata = new FormData();
  formdata.append("file", file);
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}images/upload`, {
      method: "POST",
      headers: myHeaders,
      body: formdata,
    })
    .then((response) => response);
};
