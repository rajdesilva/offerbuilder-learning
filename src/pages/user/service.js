import { baseURL } from "../../helpers";
import { userActions } from "./actions/userActions";
import { store } from "../../redux/store";

export const fetchUserInfo = () => {
  const url = `${baseURL.API_URL_BASE}user/`;
  store.dispatch({
    type: userActions.FETCH_USER_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(url)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_FAILURE,
          payload: false,
        });
      }
    })
    .catch(() => {
      store.dispatch({
        type: userActions.FETCH_USER_FAILURE,
        payload: false,
      });
    });
};

export const getUsersList = () => {
  store.dispatch({
    type: userActions.FETCH_USER_MANAGEMENT_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}users/`
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
        payload: [],
      });
    });
};

export const addNewUser = (newUserInfo) => {
  const url = `${baseURL.API_URL_BASE}users/user`;
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newUserInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const deleteUser = (email, userRole) => {
  const url = `${baseURL.API_URL_BASE}users/user/${email}?roleNames=${userRole}`;
  return window
    .fetchWrapper(url, {
      method: "DELETE",
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};
