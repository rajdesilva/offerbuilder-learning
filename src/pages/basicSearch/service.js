import { browseSupplyActions, supplySearchActions } from "./actions";
import { store } from "./../../redux/store";
import { globalActions } from "../../redux/globalActions";
import { baseURL } from "../../helpers";
import {
  createSearchQueryURL,
  getSearchParamsToFetchMarketPrice,
} from "../../helpers/utility";
import { offerListSearchAndFilterActions } from "../offers/actions";

export const searchSupply = (routePostSearch, isEditFlow) => {
  store.dispatch({
    type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_LOADING,
    payload: true,
  });
  store.dispatch({
    type: supplySearchActions.SET_INCLUDE_MARKET_PRICE_FLAG,
    payload: false,
  });

  window
    .fetchWrapper(`${baseURL.API_URL_BASE}search/` + createSearchQueryURL(routePostSearch))
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_SUCCESS,
          payload: window.getValue(response, "data.properties"),
          totalMatch: window.getValue(response, "data.totalMatchedProperties"),
          pageOffset: window.getValue(response, "data.pageOffset"),
          pageSize: window.getValue(response, "data.pageSize"),
        });
      } else {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
          payload: window.getValue(response, "error.apierror.message"),
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
        payload: error.toString(),
      });
      store.dispatch({
        type: globalActions.SET_ERROR,
        payload: error.toString(),
      });
    });
};

export const advancedSearch = (
  params,
  searchType,
  page,
  pageSize,
  sortBy,
  order,
  groupType
  ) => {
    return window.fetchWrapper(`${baseURL.API_URL_BASE}global/properties?address=${params.address}&city=${params.city}&countryCode=${params.countryCode}&distance=${params.distance}&hotelName=${params.hotelName}&latitude=${params.latitude}&longitude=${params.longitude}&zip=${params.zip}&pageOffset=${page}&pageSize=${pageSize}&propertyGroupType=${groupType}&sortCriteria=${sortBy}&sortOrder=${order}&globalSearchType=${searchType}`);

    //return window.fetchWrapper(`https://offerbuilder-api-dev.kognitiv.com/global/properties?distance=10&globalSearchType=BASIC&latitude=47.2816&longitude=15.48809&propertyGroupType=ALL`);
}

export const globalSearchResult =(params)=>{
  
  window
    .fetchWrapper(`/global/properties?address=${params.address}`)
    .then((response) => {
        console.log(response)
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
}

export const getChannels = () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_CHANNELS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}portals`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_SUCCESS,
          payload: window.getValue(response, "data.result"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getBrands = async () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_BRANDS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}sources`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_SUCCESS,
          payload: window.getValue(response, "data.brands"),
        });
        store.dispatch({
          type: offerListSearchAndFilterActions.UPDATE_OFFER_SELECTORS,
          payload: window.getValue(response, "data.brands"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_FAILURE,
          payload: [],
        });
      }
      return window.getValue(response, "data.brands");
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_BRANDS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getCurrencies = async () => {
  store.dispatch({
    type: supplySearchActions.FETCH_CURRENCIES_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}currencies`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_SUCCESS,
          payload: window.getValue(response, "data.currencies"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getMarketPricesForListOfProperties = () => {
  store.dispatch({
    type: supplySearchActions.FETCH_MARKET_PRICE_LOADING,
    payload: true,
  });

  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}search/marketprice/` +
        getSearchParamsToFetchMarketPrice()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
        payload: [],
      });
      return [];
    });
};
