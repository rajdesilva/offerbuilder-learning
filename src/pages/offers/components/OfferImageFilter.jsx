import React, { Fragment, useEffect, useState } from "react";
import { FormattedMessage, useIntl } from "react-intl";
import { Button, Collapse, Input, Modal, Badge, DatePicker, Radio } from "antd";
import moment from "moment/min/moment-with-locales";
import { FilterFilled, SearchOutlined } from "@ant-design/icons";
import { useDispatch, useSelector } from "react-redux";
import { searchAndFilterImages } from "../service";
import { isEqual } from "lodash";
import { getImageFilterCount } from "../../../helpers/utility";
import { offerImageSearchAndFilterActions } from "../actions/offerImageSearchAndFilterActions";
import { appConstants } from "../../../common";
import styles from "./css/OfferImageFilter.module.less";

const { Panel } = Collapse;

export default function OfferImageFilter(props) {
  const intl = useIntl();
  const dispatch = useDispatch();
  const [visible, setVisible] = useState(false);

  const { appliedImageFilters } = useSelector(
    (state) => window.getValue(state, "offerimagesearchandfilters"),
    isEqual
  );

  const [filterCount, setFilterCount] = useState(
    getImageFilterCount(appliedImageFilters)
  );

  // for search
  const [currentFileName, setCurrentFileName] = useState(
    appliedImageFilters.fileName
  );
  const [uploadDateRange, setUploadDateRange] = useState(
    window.getValue(appliedImageFilters, "uploadDateRange.uploadedStartDate")
      ? [
          moment(appliedImageFilters.uploadDateRange.uploadedStartDate),
          moment(appliedImageFilters.uploadDateRange.uploadedEndDate),
        ]
      : []
  );
  const [uploadedBy, setUploadedBy] = useState(
    appliedImageFilters.uploadedByCurrentUser
  );
  const [usedInOffers, setUsedInOffers] = useState(
    appliedImageFilters.usedInAnyOffer
  );
  const [offerIdSelected, setOfferIdSelected] = useState(
    appliedImageFilters.offerIds
  );

  const handleOk = () => {
    dispatch({
      type: offerImageSearchAndFilterActions.APPLY_OFFER_IMAGE_FILTERS,
      payload: {
        fileName: currentFileName,
        uploadDateRange: {
          uploadedStartDate: uploadDateRange[0]
            ? moment(uploadDateRange[0]).format("YYYY-MM-DD")
            : "",
          uploadedEndDate: uploadDateRange[1]
            ? moment(uploadDateRange[1]).format("YYYY-MM-DD")
            : "",
        },
        uploadedByCurrentUser: uploadedBy,
        usedInAnyOffer: usedInOffers,
        offerIds: offerIdSelected,
      },
    });
    dispatch({
      type: offerImageSearchAndFilterActions.RESET_OFFERS_PAGINATION,
    });
    props.hideDetailsView();
    searchAndFilterImages();
    cancelModal();
  };

  const cancelModal = () => {
    console.log("cancelModal called");
    setVisible(false);
  };

  const resetFilter = () => {
    dispatch({
      type: offerImageSearchAndFilterActions.APPLY_OFFER_IMAGE_FILTERS,
      payload: {
        fileName: "",
        uploadDateRange: {
          uploadedStartDate: "",
          uploadedEndDate: "",
        },
        uploadedByCurrentUser: appConstants.OFFER_IMAGE.ALL,
        usedInAnyOffer: appConstants.OFFER_IMAGE.ALL,
        offerIds: "",
      },
    });
    dispatch({
      type: offerImageSearchAndFilterActions.RESET_OFFERS_PAGINATION,
    });
    props.hideDetailsView();
    searchAndFilterImages();
    cancelModal();
  };

  const onSearchFileName = (e) => {
    console.log("onSearchFileName =", e.target.value);
    setCurrentFileName(e.target.value);
  };

  const onDateSelected = (dates) => {
    console.log("onDateSelected =", dates);
    if (dates) {
      if (dates[0] && dates[1]) {
        setUploadDateRange(dates);
      }
    } else {
      setUploadDateRange([]);
    }
  };

  const onOfferIdSelected = (e) => {
    console.log("onOfferIdSelected =", e.target.value);
    setOfferIdSelected(e.target.value);
  };

  const onUploadedBySelected = (e) => {
    console.log("onUploadedBySelected =", e.target.value);
    setUploadedBy(e.target.value);
  };

  const onUsedInOfferSelected = (e) => {
    console.log("onUsedInOfferSelected =", e.target.value);
    setUsedInOffers(e.target.value);
  };

  useEffect(() => {
    setFilterCount(getImageFilterCount(appliedImageFilters));
    setCurrentFileName(appliedImageFilters.fileName);
    setUploadDateRange(
      window.getValue(appliedImageFilters, "uploadDateRange.uploadedStartDate")
        ? [
            moment(appliedImageFilters.uploadDateRange.uploadedStartDate),
            moment(appliedImageFilters.uploadDateRange.uploadedEndDate),
          ]
        : []
    );
    setUploadedBy(appliedImageFilters.uploadedByCurrentUser);
    setUsedInOffers(appliedImageFilters.usedInAnyOffer);
    setOfferIdSelected(appliedImageFilters.offerIds);
  }, [appliedImageFilters]);

  return (
    <Fragment>
      <div onClick={() => setVisible(true)} className={styles.badgeContainer}>
        <Badge
          count={filterCount}
          offset={[-5, 5]}
          size="small"
          data-testid="offers-image-filter-badge"
          className={styles.filterBadge}
        >
          <Button
            type="text"
            data-testid="filter-ic-btn"
            icon={<FilterFilled size={20} className='filter-ic' />}
          ></Button>
        </Badge>
      </div>
      <Modal
        title={
          <React.Fragment>
            <FormattedMessage id="nemo.filterBy" />
          </React.Fragment>
        }
        onCancel={cancelModal}
        visible={visible}
        closable={true}
        width={330}
        className={styles.filterModal}
        destroyOnClose={true}
        forceRender={true}
        data-testid="offer-image-filter-modal"
        okText={<FormattedMessage id="nemo.yes" />}
        cancelText={<FormattedMessage id="nemo.cancel" />}
        footer={[
          <Button
            key="cancel"
            onClick={cancelModal}
            data-testid="filter-offer-image-cancel-btn"
          >
            <FormattedMessage id="nemo.cancel" />
          </Button>,
          <Button
            key="reset"
            onClick={resetFilter}
            data-testid="filter-offer-image-reset-btn"
          >
            <FormattedMessage id="nemo.reset" />
          </Button>,
          <Button
            key="submit"
            type="primary"
            onClick={handleOk}
            data-testid="filter-offer-image-apply-btn"
          >
            <FormattedMessage id="nemo.apply" />
          </Button>,
        ]}
      >
        <Collapse
          defaultActiveKey={["filename"]}
          ghost
          expandIconPosition="right"
          style={{ backgroundColor: "transparent" }}
        >
          <Panel
            header={<FormattedMessage id="nemo.filename" />}
            key="filename"
          >
            <Input
              placeholder={intl.formatMessage({
                id: "nemo.mediumInput",
              })}
              size="middle"
              allowClear
              value={currentFileName}
              prefix={<SearchOutlined className={styles.searchIcon} />}
              onChange={onSearchFileName}
            />
          </Panel>
          <Panel
            header={<FormattedMessage id="nemo.uploadDate" />}
            key="upload-date"
          >
            <DatePicker.RangePicker
              data-testid={"upload-range-picker"}
              format={moment.localeData().longDateFormat("L")}
              defaultValue={uploadDateRange}
              onCalendarChange={onDateSelected}
            />
          </Panel>
          <Panel
            header={<FormattedMessage id="nemo.usedInOffers" />}
            key="used-in-offers"
          >
            <Fragment>
              <Radio.Group
                onChange={onUsedInOfferSelected}
                value={usedInOffers}
              >
                <Radio value={appConstants.OFFER_IMAGE.ALL}>
                  <FormattedMessage id="nemo.yes" />
                </Radio>
                <Radio value={appConstants.OFFER_IMAGE.EXCLUDE}>
                  <FormattedMessage id="nemo.no" />
                </Radio>
              </Radio.Group>
            </Fragment>
          </Panel>
          <Panel
            header={<FormattedMessage id="nemo.uploadedByUser" />}
            key="uploaded-by-user"
          >
            <Fragment>
              <Radio.Group onChange={onUploadedBySelected} value={uploadedBy}>
                <Radio value={appConstants.OFFER_IMAGE.INCLUDE}>
                  <FormattedMessage id="nemo.currentUser" />
                </Radio>
                <Radio value={appConstants.OFFER_IMAGE.ALL}>
                  <FormattedMessage id="nemo.everyone" />
                </Radio>
              </Radio.Group>
            </Fragment>
          </Panel>
          <Panel
            header={<FormattedMessage id="nemo.offersId" />}
            key="offers-id"
          >
            <Input
              placeholder={intl.formatMessage({
                id: "nemo.enterId",
              })}
              allowClear
              size="middle"
              value={offerIdSelected}
              prefix={<SearchOutlined className={styles.searchIcon} />}
              onChange={onOfferIdSelected}
            />
          </Panel>
        </Collapse>
      </Modal>
    </Fragment>
  );
}
