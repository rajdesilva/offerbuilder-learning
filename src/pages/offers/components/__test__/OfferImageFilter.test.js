import React from "react";
import { createStore } from "redux";
import {
  render,
  fireEvent,
  cleanup,
  waitFor,
} from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import OfferImageFilter from "../OfferImageFilter";
import { tempStore } from "./tempStore";
import { act } from "react-dom/test-utils";
import * as service from "./../../service";
import { appConstants } from "../../../../common";

import { cloneDeep } from "lodash";

window.fetchWrapper = window.fetchWrapper = jest.fn().mockResolvedValue({
  success: true,
});
jest.mock("./../../service");
afterEach(cleanup);
describe("OfferImageFilter component test", () => {
  test("create OfferImageFilter components snapshot", async () => {
    const filterStore = cloneDeep(tempStore);
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { container, getByTestId } = render(
      <MemoryRouter>
        <OfferImageFilter />
      </MemoryRouter>,
      {
        store,
      }
    );

    const filterBtn = getByTestId("filter-ic-btn");
    act(() => {
      fireEvent.click(filterBtn);
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  test("OfferImageFilter components- click on cancel ", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      hideDetailsView: jest.fn(),
    }
    const { getByTestId, queryByText } = render(
      <MemoryRouter>
        <OfferImageFilter {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const filterBtn = getByTestId("filter-ic-btn");
    act(() => {
      fireEvent.click(filterBtn);
    });
    // screen.debug();
    const cancelBtn = getByTestId("filter-offer-image-cancel-btn");
    act(() => {
      fireEvent.click(cancelBtn);
    });
    waitFor(() => {
      expect(queryByText("Filter By")).not.toBeInTheDocument();
    })
  });

  test("OfferImageFilter components- click on reset and see dispatch is called", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      hideDetailsView: jest.fn(),
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferImageFilter {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const filterBtn = getByTestId("filter-ic-btn");
    await act(async () => {
      await fireEvent.click(filterBtn);
    });
    const resetBtn = getByTestId("filter-offer-image-reset-btn");
    act(() => {
      fireEvent.click(resetBtn);
    });
    expect(service.searchAndFilterImages).toHaveBeenCalledTimes(1);
    expect(props.hideDetailsView).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledTimes(2);
  });

  test("OfferImageFilter components- click on apply and see if service is called", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      hideDetailsView: jest.fn(),
    }
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferImageFilter {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const filterBtn = getByTestId("filter-ic-btn");
    await act(async () => {
      await fireEvent.click(filterBtn);
    });
    const applyBtn = getByTestId("filter-offer-image-apply-btn");
    act(() => {
      fireEvent.click(applyBtn);
    });
    expect(store.dispatch).toHaveBeenCalledTimes(2);
  });

  test("OfferImageFilter components- check if badge is in the DOM or not and counter is hidden for initial case", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      hideDetailsView: jest.fn(),
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferImageFilter {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const filterBtn = getByTestId("filter-ic-btn");
    act(() => {
      fireEvent.click(filterBtn);
    });
    const badge = getByTestId("offers-image-filter-badge");
    expect(badge).toBeTruthy();
    const counter = badge.querySelector(
      '[class="ant-scroll-number-only-unit current"]'
    );
    expect(counter).toBe(null);
  });

  test("OfferImageFilter components- check if badge is in the DOM with filter selected count as 1", async () => {
    const filterStore = cloneDeep(tempStore);
    filterStore.offerimagesearchandfilters.appliedImageFilters.fileName =
      "testfile";
    const store = createStore(() => ({ ...filterStore }));
    store.dispatch = jest.fn();
    const props = {
      hideDetailsView: jest.fn(),
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferImageFilter {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const filterBtn = getByTestId("filter-ic-btn");
    act(() => {
      fireEvent.click(filterBtn);
    });
    const badge = getByTestId("offers-image-filter-badge");
    const counter = badge.querySelector(
      '[class="ant-scroll-number-only-unit current"]'
    );
    expect(counter.textContent).toBe("1");
  });

  test("OfferImageFilter components- check if badge is in the DOM with count as 5 for filter selected brand, storefront, supplier, channel", async () => {
    const filterStore = cloneDeep(tempStore);
    filterStore.offerimagesearchandfilters.appliedImageFilters = {
      fileName: "testing",
      uploadDateRange: {
        uploadedStartDate: "2020-01-01",
        uploadedEndDate: "2021-03-01",
      },
      uploadedByCurrentUser: appConstants.OFFER_IMAGE.INCLUDE,
      usedInAnyOffer: appConstants.OFFER_IMAGE.EXCLUDE,
      offerIds: "offer123",
    };
    const props = {
      hideDetailsView: jest.fn(),
    };
    const store = createStore(() => ({ ...filterStore }));
    store.dispatch = jest.fn();
    const { getByTestId } = render(
      <MemoryRouter>
        <OfferImageFilter {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    const filterBtn = getByTestId("filter-ic-btn");
    act(() => {
      fireEvent.click(filterBtn);
    });
    const badge = getByTestId("offers-image-filter-badge");
    const counter = badge.querySelector(
      '[class="ant-scroll-number-only-unit current"]'
    );
    expect(counter.textContent).toBe("5");
  });
});
