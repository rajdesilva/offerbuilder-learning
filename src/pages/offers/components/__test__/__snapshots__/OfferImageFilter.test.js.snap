// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferImageFilter component test create OfferImageFilter components snapshot 1`] = `
<div>
  <span
    class="ant-badge"
    data-testid="offers-image-filter-badge"
  >
    <button
      class="ant-btn ant-btn-text ant-btn-icon-only"
      data-testid="filter-ic-btn"
      type="button"
    >
      <span
        aria-label="filter"
        class="anticon anticon-filter filter-ic"
        role="img"
        size="20"
      >
        <svg
          aria-hidden="true"
          data-icon="filter"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
          />
        </svg>
      </span>
    </button>
  </span>
</div>
`;
